data-generation/src/main/scala/LogTSV.scala

```diff
@@ -0,0 +1,176 @@
+}
+
+/** LogTSV
+ *
+ * Everything is built around this case class
+ * @param timestamp When did the log_entry happen?
+ * @param action Type of action
+ * @param objectId Id of the object
+ * @param attributes List of (Key,Value) attributes relevant to the entry
+ */
+case class LogTSV(
+                 timestamp : Instant,
+                 action : Action,
+                 objectId: Long,
+                 attributes: Attributes
+                 )
+
+object LTSV {
+
+  /** Random LogTSV instance
+   *
+   * Convenient for testing
+   * @return An instance of LogTSV
+   */
+  def randomLTSV():LogTSV = {
+    val actions = List(CREATE, UPDATE, DELETE)
+    val faker = new Faker()
+    LogTSV(
+      timestamp = Instant.now(),
+      action = actions(nextInt(actions.size)),
+      objectId = nextInt(10000),
+      attributes = List(("champion",faker.leagueOfLegends().champion()),
+        ("friends", faker.friends().character()))
+    )
+
+
+
+  }
+  // Type alias
+  type Attributes = List[(String, String)]
+
+  /** Serialize a single LogTSV
+   *
+   * @param log_entry The log entry
+   * @return String representation of the LogTSV
+   */
+  def serializeLTSV(log_entry:LogTSV ): String ={
+    val timestamp = LocalDateTime.ofInstant(log_entry.timestamp, ZoneOffset.UTC).toString +"Z"
+    val action = log_entry.action match {
+      case CREATE => "CREATE"
+      case UPDATE => "UPDATE"
+      case DELETE => "DELETE"
+    }
+    val objectId = log_entry.objectId.toString
+    val attributes = serializeAttributes(log_entry.attributes)
+
+    List(timestamp, action, objectId, attributes)
+      .mkString("\t")
+  }
+
+  def serializeList(entries: List[LogTSV]):String = entries.map(serializeLTSV).mkString("\n")
+  def deserializeList(entries: String):List[LogTSV] = entries.split("\n").flatMap(deserializeLTSV).toList
+
+  /**  Deserialize a single LogTSV
+   *
+   * Check this shit out. Monadic error handling
+   *
+   * All the variables with s_ prefix return an Option type.
+   * If any return a None the whole block returns a None.
+   * If every one of the are parsed correctly it return Some(LogTsv)
+   *
+   * TODO Rewrite this into Either-type so that information on what is going wrong isn't lost
+   *
+   * @param log_entry String representation of a log entry
+   * @return Possibly a LogTSV
+   */
+  def deserializeLTSV(log_entry:String):Option[LogTSV] = {
+    // This line could fail, but its a hassle to make safe
+    val timestamp::action::objectId::attributes : List[String]= log_entry.split('\t').toList
+    for {
+      s_timestamp <- Try {Instant.parse(timestamp)}.toOption
+      s_action <- action match {
+        case "CREATE" => Some(CREATE)
+        case "UPDATE" => Some(UPDATE)
+        case "DELETE" => Some(DELETE)
+        case _ => None
+      }
+      s_objectId <- Try{objectId.toLong}.toOption
+      s_attributes <- deserializeAttributes(attributes)
+    } yield LogTSV(s_timestamp, s_action, s_objectId, s_attributes)
+  }
```

Comment:
Looks cool, although I'm not that familiar with Scala syntax yet 😅 
Is the `s_` prefix an abbreviation for something? And is there a reason for combined snake case and camel case? 🐍 🐫 

Some formatting and casing conventions
```suggestion
  /**  Deserialize a single LogTSV
   * All the variables with s_ prefix return an Option type.
   * If any return a None the whole block returns a None.
   * If every one of the are parsed correctly it return Some(LogTsv)
   *
   * TODO Rewrite this into Either-type so that information on what is going wrong isn't lost
   *
   * @param logEntry String representation of a log entry
   * @return Possibly a LogTSV
   */
  def deserializeLTSV(logEntry: String): Option[LogTSV] = {
    // This line could fail, but its a hassle to make safe
    val timestamp::action::objectId::attributes : List[String] = logEntry.split('\t').toList
    for {
      s_timestamp <- Try {Instant.parse(timestamp)}.toOption
      s_action <- action match {
        case "CREATE" => Some(CREATE)
        case "UPDATE" => Some(UPDATE)
        case "DELETE" => Some(DELETE)
        case _ => None
      }
      s_objectId <- Try{objectId.toLong}.toOption
      s_attributes <- deserializeAttributes(attributes)
    } yield LogTSV(s_timestamp, s_action, s_objectId, s_attributes)
  }
```

Reply:
jeg er veldig dårlig på case convetions. Glemmer hele tiden hva som er vanlig i Scala haha. Skal fikse det. s_ prefixen er for serialized. Men burde kanskje være d_ for deserialized. Altså s_timestamp er en deserialized timestamp.

Reply:
vanskelig å holde tunga rett i munn

Reply:
Kan vi skrive det rett ut, eks. `serializedTimestamp`? Er med på forkortelsen hvis vi gjør det til et felt/member på et objekt, men innad i scopet her vil jeg si det hjelper en del på lesbarheten

Reply:
for later reference: https://docs.scala-lang.org/style/naming-conventions.html
ja enig, tydeligere det. Sjekker det ut i mrg